src/MoBi.Presentation/Presenter/EditSimulationPresenter.cs

```diff
@@ -328,11 +328,23 @@ private bool canHandle(IObjectBaseEvent objectBaseEvent)
 
       private void onObservedDataAddedToChart(object sender, ObservedDataAddedToChartEventArgs e)
       {
-         foreach (var dataRepository in e.AddedDataRepositories)
+         var observedDataToAdd = observedDataNotMappedInTheSimulation(e);
+
+         if (!observedDataToAdd.Any())
+            return;
+
+         foreach (var dataRepository in observedDataToAdd)
          {
             _outputMappingMatchingTask.AddMatchingOutputMapping(dataRepository, _simulation);
             _context.PublishEvent(new ObservedDataAddedToAnalysableEvent(_simulation, dataRepository, false));
          }
       }
+
+      private List<DataRepository> observedDataNotMappedInTheSimulation(ObservedDataAddedToChartEventArgs e)
+      {
+         var observedDataToAdd = e.AddedDataRepositories
+            .Where(dataRepository => !_simulation.OutputMappings.Any(x => x.UsesObservedData(dataRepository))).ToList();
+         return observedDataToAdd;
```

Comment:
You can just return right away without the variable

Reply:
true. corrected.